controller/konnect/ops/ops_kongpluginbinding.go

```diff
@@ -46,16 +46,16 @@ func createPlugin(
 		*pluginInput,
 	)
 
-	// TODO: handle already exists
-	// Can't adopt it as it will cause conflicts between the controller
-	// that created that entity and already manages it, hm
-	if errWrap := wrapErrIfKonnectOpFailed[configurationv1alpha1.KongPluginBinding](err, CreateOp, pb); errWrap != nil {
-		SetKonnectEntityProgrammedConditionFalse(pb, "FailedToCreate", errWrap.Error())
-		return errWrap
+	if errWrapped := wrapErrIfKonnectOpFailed(err, CreateOp, pluginBinding); errWrapped != nil {
+		return errWrapped
 	}
 
-	pb.SetKonnectID(*resp.Plugin.ID)
-	SetKonnectEntityProgrammedCondition(pb)
+	if resp == nil || resp.Plugin == nil || resp.Plugin.ID == nil {
+		return fmt.Errorf("failed creating %s: %w", pluginBinding.GetTypeName(), ErrNilResponse)
+	}
+
+	pluginBinding.SetKonnectID(*resp.Plugin.ID)
+	SetKonnectEntityProgrammedCondition(pluginBinding)
```

Comment:
We don't want to set the conditions in side the type specific ops* functions.

This is now handled in https://github.com/Kong/gateway-operator/blob/91cd7ff2a0c12ea5fb7bc0d1a920276875d41746/controller/konnect/ops/ops.go#L151-L163

Reply:
Fixed in https://github.com/Kong/gateway-operator/pull/772